### PR TITLE
release-0.66 (closed)

### PR DESCRIPTION
**Pull request by TwitchBronBron**
Pull request for v0.66.0, which includes some breaking changes.
- new [type tracking](https://github.com/rokucommunity/brighterscript/pull/783) improvements, but comes with some breaking changes.

### CODE REVIEW

File: src/bscPlugin/CallExpressionInfo.ts [hunk 1]
```diff
@@ -66,11 +66,11 @@ export class CallExpressionInfo {
             this.newExpression = callExpression.parent;
         }
         if (isCallfuncExpression(callExpression)) {
-            this.name = callExpression.methodName.text;
+            this.name = (callExpression as CallfuncExpression).methodName.text;
```
Comment (elsassph) [inline src/bscPlugin/CallExpressionInfo.ts:69]: What is happening here that we need this `as`?

File: src/parser/Statement.ts [hunk 2]
```diff
@@ -1132,7 +1193,7 @@ export class NamespaceStatement extends Statement implements TypedefProvider {
     constructor(
         public keyword: Token,
         // this should technically only be a VariableExpression or DottedGetExpression, but that can be enforced elsewhere
-        public nameExpression: NamespacedVariableNameExpression,
+        public nameExpression: Expression,
```
Comment (elsassph) [inline src/parser/Statement.ts:1196]: Would prefer seeing it narrowed here as A | B type maybe

File: src/bscPlugin/CallExpressionInfo.ts (hunk 1 above)
Comment (markwpearce) [reply src/bscPlugin/CallExpressionInfo.ts]: I think this was introduced because of the reflection changes. actual types were updated to resolve this

**Comment by TwitchBronBron** [comment]
Closing this branch in favor of the new v1.0.0 branch and PR (https://github.com/rokucommunity/brighterscript/pull/1053)

**Pull request closed without merging**